python/ray/data/dataset.py

```diff
@@ -877,6 +877,16 @@ def split(
         if n <= 0:
             raise ValueError(f"The number of splits {n} is not positive.")
 
+        if equal:
```

Comment:
we are ignoring `locality_hints` here. @ericl @matthewdeng remind me what's our decision to notify our user?

Reply:
Can we only do this if `locality_hints` aren't specified?

Reply:
Do we really need locality_hints for current SSML usecases?
With the split_at_indice implementation which doesn't respect locality_hints, for each splitted dataset shard, if it contains N blocks, only constant number of blocks (at most 2) will be created and misplaced; the remaining N-2 blocks will not being processed/moved anyway.

Reply:
Why don't we (1) not pass locality from AIR when calling split for now, and (2) document locality hints as experimental? I think this would be most understandable from the user perspective.

Reply:
sgtm